lib/webmock/util/uri.rb

```diff
@@ -47,7 +47,7 @@ def self.variations_of_uri_as_strings(uri_object)
           uris = uris_with_inferred_port_and_without(uris)
         end
 
-        if normalized_uri.scheme == "http"
+        if normalized_uri.scheme == "http" && !with_scheme
```

Comment:
I'm not sure I understand the with_scheme flag here. The condition is "if not with scheme" and the line below compares returns values with scheme and without.

Reply:
The flag means "returns only variations with scheme", but I agree it's confusing.

`only_variations_with_scheme` might be better?

Reply:
Yes `only_with_scheme` would be more meaningful.

Reply:
updated https://github.com/bblimke/webmock/pull/758/commits/19624000ba9701ade80df97a5d92d3bc3c2ca4a6